UPDATE README (I eventually figured out that it is a "Server-side API" )

Pierreparis opened the pull request:
If you wish to spare time of other programmers, you should state (Clearly) in the first place that this JavaScript library work on server side because  JavaScript being used most on client side we can't immediately understand especially if you assume that the people don't all speak a perfect English, and that some other JavaScript API like the Soundcloud API work the other way around on client side.

//PLEASE Change this : "JavaScript client library for the Tumblr API"
//TO THIS  "JavaScript (not really client library  because it work on the server) for the Tumblr API"

seejohnrun committed:
Fix package.json

seejohnrun committed:
Update README for real NPM name

seejohnrun committed:
Added README link to NPM

seejohnrun committed:
Travis

seejohnrun committed:
Update README

seejohnrun committed:
Some space

seejohnrun committed:
Fix for userLikes

seejohnrun committed:
Bump version to 0.0.2

seejohnrun committed:
Fix to package.json

seejohnrun committed:
Bump version to 0.0.3

seejohnrun committed:
Added README example

oleglitvin committed:
Update README.md

fix typo

seejohnrun committed:
Merge pull request #1 from OlegLitvin/patch-1

Update README.md

seejohnrun committed:
Change how params are passed into request on _get

Closes #2

seejohnrun committed:
Bump version to 0.0.4

seejohnrun committed:
Fix test

seejohnrun committed:
Browser include fix

daltomania committed:
add missing semicolon; add to dev dependencies and update makefile

Commit:
Check for body then check if body has a response (separate tests yielded secondary error)

seejohnrun committed:
Merge pull request #4 from daltomania/tiny-fix

add missing semicolon; add to dev dependencies and update makefile

seejohnrun committed:
fix a spec for new version of should

seejohnrun committed:
Only parse body on non-error for POST

Closes #8

seejohnrun committed:
Fix for malformed body in POST

[#5]

seejohnrun committed:
Merge pull request #7 from adrianseeley/patch-3

Check for body then check if body has a response (separate tests yielded...

seejohnrun committed:
Some code cleanup

seejohnrun committed:
Fix some lint issues

seejohnrun committed:
Expand README with more notes

Closes #10

seejohnrun committed:
Fix a README inconsistency

jhorbulyk committed:
Update Documentation: Error in example.

seejohnrun committed:
Merge pull request #13 from jhorbulyk/master

Update Documentation: Error in example.

Commit:
updated the library to send the user agent with it

Commit:
updated library version to read from package.json file

Commit:
updated package.json file to be correct version

ngreenwood6 committed:
Merge pull request #21 from tumblr/likes-before-after

Added report versioning to the api

codingjester committed:
Adding CONTRIBUTING.md

irace commented:
> "JavaScript (not really client library because it work on the server) for the Tumblr API"

It doesn’t only work on the server. You could write a script that you run with node on your local machine. “Client” doesn’t mean browser, it means client in the [client-server model](http://en.wikipedia.org/wiki/Client–server_model).

seejohnrun commented:
Yep - and this is actually used client-side in `api.tumblr.com/console`

Pierreparis commented:
Ok sorry and I didn't see it.

roderickhsiao committed:
Loose request version

ceyko committed:
Merge pull request #35 from roderickhsiao/patch-1

Loose request version

ceyko committed:
Version bump to v0.0.7

Previously forgot to version bump to v0.0.6 before making a release, so
go directly to v0.0.7 for the next public release.

jlobos committed:
Fix typo

komapa committed:
Merge pull request #45 from jlobos27/master

Fix typo

kmck committed:
tumblr.js v1.0.0

* rewrote api client
* removed repl
* new test suite
* eslint
* updated documentation

kmck committed:
stricter eslint rules

kmck committed:
fix github urls

kmck committed:
cleanup

kmck committed:
pass full response to callback as third argument

kmck committed:
add istanbul for coverage

kmck committed:
dependency update

kmck committed:
Merge pull request #50 from kmck/kmck/tumblr.js-1.0.0

tumblr.js v1.0.0

kmck committed:
add Promise support

kmck committed:
add tests for returnPromises

kmck committed:
update README

kmck committed:
bump version to 1.1.0

kmck committed:
use options object and update tests

kmck committed:
clean up jsdocs and update readme

kmck committed:
Merge pull request #52 from tumblr/kmck/promises

Add Promise support to requests

kmck committed:
jsdoc cleanup

kmck committed:
update dependencies, add jsdoc and hotdoc

kmck committed:
generate docs

Commit:
Add link to the client documentation 

We should link out to the beautiful documentation that @kmck set up!

kmck committed:
update dependencies, skip lint on old node

Commit:
Merge pull request #54 from heylookltsme/patch-1

Add link to the client documentation

kmck committed:
version bump 1.1.1

kmck committed:
Merge pull request #55 from tumblr/kmck/fix-tests

Update dependencies, skip lint on old node

notjosh committed:
[docs] Fixed title: createLinkPost -> createChatPost

jasonpenny committed:
Merge pull request #61 from notjosh/patch-1

[docs] Fixed title: createLinkPost -> createChatPost

jasonpenny committed:
Regenerate docs

jasonpenny committed:
Allow sending multiple images in a photo post

jasonpenny committed:
Merge pull request #63 from tumblr/allow-multiple-images

Allow sending multiple images in a photo post

jasonpenny committed:
Start of typescript type definitions

jasonpenny committed:
Bump package version to 1.1.2

bjornstar committed:
Update dependencies, regenerate docs, remove gulp, update node version, update README

ceyko committed:
Merge pull request #70 from bjornstar/bump-dependencies

Update dependencies

jasonpenny committed:
Merge branch 'master' into start-typescript-types

ceyko committed:
Merge pull request #66 from tumblr/start-typescript-types

Start of typescript type definitions

ceyko committed:
Bump version to 2.0.0

As mentioned in #67, since we've dropped support for a (very) old
version of node - 0.12, we should bump the major version.

See: https://github.com/tumblr/tumblr.js/pull/67

ceyko committed:
Merge pull request #74 from tumblr/ceyko/bump-version-2.0.0

Bump version to 2.0.0

adamgauthier committed:
avoid redeclaration of client in tests

adamgauthier committed:
use block-scoped variable declarations everywhere

john-raymon committed:
added somethings for clarity

it may be useful to use `blogName` throughout README rather than `staff`, it took me sometime to find out staff meant staff.tumblr.com. Also, it took me sometime to figure out how tags should be shaped in the options object.

john-raymon committed:
Merge pull request #1 from john-raymon/clarity-changes

added somethings for clarity

jasonpenny committed:
Merge pull request #78 from john-raymon/master

added somethings for clarity

Commit:
Fix method descriptions

jasonpenny committed:
Merge remote-tracking branch 'pimuzzo/fix-methods-descriptions'

jasonpenny committed:
Fix doc method name for blogFollowers

jasonpenny committed:
Regenerate docs

defect committed:
Update CLA information

defect committed:
Merge pull request #82 from tumblr/cla_update

Update CLA information

Commit:
Update some docs

shahkashani committed:
Merge pull request #76 from louistio/master

Use block-scoped variable declarations everywhere

shahkashani committed:
Merge pull request #83 from shahkashani/shah/update-date

Update some docs

Commit:
Update some dependencies

shahkashani committed:
Merge pull request #84 from shahkashani/shah/update-libs

Update some dependencies

scriptype committed:
Bump dependency version of request

shahkashani committed:
Merge pull request #86 from scriptype/fix-unsecure-request-dependency

Bump dependency version of request

Commit:
Bump version to reflect bump in Request

shahkashani committed:
Merge pull request #87 from shahkashani/shah/bump-patch

Bump version to reflect bump in Request

cyle committed:
compatibility with links objects

cyle committed:
remove it here too

cyle committed:
better comments

cyle committed:
update base url and endpoints

cyle committed:
ah yes, tests

cyle committed:
add helpful warning

cyle committed:
disable no-console for here only

cyle committed:
fix test

cyle committed:
Merge pull request #100 from cyle/cyle/links-compatibility

Add compatibility with Tumblr API links objects being passed through to these low level functions

sndsgd committed:
remove contributor license agreement (#102)

gravityrail committed:
Add Typescript signature for TumblrClient.getRequest

This hit me when I was writing an internal tool in typescript and wanted to access a custom endpoint.

gravityrail committed:
Add postRequest method for Typescript

sanmai committed:
Add YAML validation to bootstrap GitHub workflows

cyle committed:
Merge pull request #106 from gravityrail/patch-1

Add Typescript signature for TumblrClient.getRequest

cyle committed:
Merge pull request #107 from sanmai/pr/2021-12/ci-start

Add YAML validation to bootstrap GitHub workflows

sanmai committed:
Add testing workflow, nuke .travis.yml (#108)

rosano committed:
Fix typos (#117)

sirreal committed:
Update README build status badge

sirreal committed:
Add prettier and autoformat script
Add format-check script and CI formatting check
Update eslint config
Use eslint-config-prettier to allow prettier formatting
Fix lints

sirreal committed:
Update deps, supported node versions, switch to doccash

sirreal committed:
Update gh-pages build

sirreal committed:
Update scripts and workflows

sirreal committed:
Add CHANGELOG

sirreal committed:
Flag gh-pages as generated

sirreal committed:
Autoformat

sirreal committed:
Add .git-blame-ignore-revs file (#160)

Add .git-blame-ignore-revs to ignore certain commits from blame.

We'll use this to ignore e.g. generating gh-pages build or autoformatting commits.

sirreal committed:
Add integration tests (#161)

* Add integration tests
* Add changelog entry
* Add CONTRIBUTING section about tests
* Add ESM import support to package

sirreal committed:
Drop special handling for blogIdentifier parameters (#162)

Drop special handling for `:blogIdentifier` path parameters.
Add integration test for 3 forms of blogIdentifier path parameters.

The API documentation is clear that three forms are allowed.
See https://www.tumblr.com/docs/en/api/v2#blog-identifiers.

- Blog name e.g. staff for https://staff.tumblr.com/
- Hostname e.g. staff.tumblr.com
- Universally unique identifier (UUID) e.g. t:0aY0xL2Fi1OFJg4YxpmegQ for staff

Special treatment of blogIdentifiers was buggy and broke the UUID form.
It was unnecessary, the API handles blog name or hostname properly
(`staff` is treated the same as `staff.tumblr.com`).

Closes #93

Co-authored-by: vipulrawat <vipulrawat.imi@live.in>

sirreal committed:
Refactor as class and simplify options (#163)

- Refactor as class
- Remove multiple argument constructor form
- Credentials should be provided in options.
- Remove `request` option
- Update tests to remove request option
- Update type declarations with Options

sirreal committed:
Refactor for v4.0.0 (#153)

This is a significant rewrite of the library in order to remove the deprecated request dependency (#91).

A number of other changes are made in order to make the library easier to maintain and work with in the future. I've done my best to balance breaking changes and deprecations with maintaining a reasonable upgrade path for existing users.

Detailed changes are described in the CHANGELOG.
